graphics.cpp

```diff
@@ -67,3 +46,35 @@ void Graphics::AddSprite(const Sprite& sprite, const SDL_Rect& destination,
   SDL_RenderCopy(renderer_, sprite.GetSpriteTexture(), &source, &destination);
   perf->StopTimer("render_copy");
 }
+
+void Graphics::AddSprite(AnimatedSprite& sprite, Perf* perf,
+                         std::string animation) {
+  if (sprite.GetVisible()) {
+    int frame = sprite.GetFrameIndex();
+    SDL_Rect source = sprite.GetSourceRect(animation, frame);
+    SDL_Rect destination = sprite.GetDestinationRect();
```

Comment:
The sprite knows the destination, and the sprite knows the offsets for the given animation, and the sprite needs the new destination rect. This is all AnimatedSprite logic leaking into the graphics class.

Maybe add a new method in animated sprite called "Animate(std::string animation)", and just do the computation inside the sprite class, updating the destination rect internally.